### PR TITLE
Remove space id from all database table except space table

### LINKED ISSUE #14: Tech: Delete Product should cascade and delete assignments using DB
When we delete a product we want the SQL database to take care of the deletion of assignments

**Current state**: When we delete a product, the assignment service is called to delete on the assignments for first. 

**Desired State**: When deleting a product, the assignments should be deleted. We should be having the SQL database doing it for us.

### PR DESCRIPTION
**Pull request by willot**
## Issue
Resolves #14 part 2

## What was done
- [x] Part 1 was adding the uuid to the tables that needed it
- [x] Part 2 is removing the spaceId from the table and make sure everything use the space uuid instead 
- [x] Product tag end point now verify that the product tag deleted or updated belong to the space that the user send with the product tag Id

## How to test
1. Everything should work the same

### CODE REVIEW

File: api/src/main/kotlin/com/ford/internalprojects/peoplemover/producttag/ProductTagRepository.kt [hunk 1]
```diff
@@ -24,5 +24,8 @@ import org.springframework.stereotype.Repository
 @Repository
 interface ProductTagRepository : PeopleMoverRepository<ProductTag, Int> {
     fun findAllByNameIgnoreCaseAndSpaceUuid(name: String, spaceUuid: String): ProductTag?
+    fun findAllByNameAndSpaceUuid(name: String, spaceUuid: String): ProductTag?
     fun findAllBySpaceUuid(spaceUuid: String, name: Sort): List<ProductTag>
+    fun findByIdAndSpaceUuid(id: Int, spaceUuid: String): ProductTag?
```
Comment (dulcimerist) [inline api/src/main/kotlin/com/ford/internalprojects/peoplemover/producttag/ProductTagRepository.kt:29]: Is this a product ID? A space ID?

File: api/src/main/kotlin/com/ford/internalprojects/peoplemover/space/Space.kt [hunk 2]
```diff
@@ -39,12 +36,6 @@ data class Space (
     @Column(unique = true, nullable = false)
     var name: String,
 
-    @OneToMany(mappedBy = "spaceId", orphanRemoval = true, cascade = [CascadeType.REMOVE, CascadeType.REFRESH], fetch = FetchType.EAGER)
-    val roles: Set<SpaceRole> = HashSet(),
-
-    @OneToMany(mappedBy = "spaceId", orphanRemoval = true, cascade = [CascadeType.REMOVE, CascadeType.REFRESH], fetch = FetchType.EAGER)
-    val locations: List<SpaceLocation> = ArrayList(),
-
```
Comment (dulcimerist) [inline api/src/main/kotlin/com/ford/internalprojects/peoplemover/space/Space.kt:47]: Did we ever replace these with one-to-many mappings for spaceUuid? I see in the repository layer we're now exposing a "findAllBySpaceUuid" function - is that a comprehensive replacement for getting rid of these annotations?

Comment (alexwojtala) [reply api/src/main/kotlin/com/ford/internalprojects/peoplemover/space/Space.kt:47]: We did not replace these and yes, we use findAllBySpaceUuid instead toe get the locations and roles associated with a space

**Review by aclift19Ford — APPROVED** [review]
Just a couple things I noticed, but everything seems good to Catkin and I otherwise :)

**Review by alexwojtala — APPROVED** [review]

File: api/src/test/kotlin/com/ford/internalprojects/peoplemover/role/RoleControllerApiTest.kt [hunk 3]
```diff
@@ -143,7 +143,7 @@ class RoleControllerApiTest {
     @Test
     fun `GET should return 200 when valid token that isn't an editor requests a space while read-only flag is on`() {
         val anonymousUserReadOnlySpace: Space = spaceRepository.save(Space(name = "SpaceOne", todayViewIsPublic = true))
-        val role1: SpaceRole = spaceRolesRepository.save(SpaceRole(name = "Fireman", spaceId = anonymousUserReadOnlySpace.id!!, spaceUuid = space.uuid))
+        val role1: SpaceRole = spaceRolesRepository.save(SpaceRole(name = "Fireman", spaceUuid = space.uuid))
```
[the remark this answers is not in the document]
Comment (alexwojtala) [reply api/src/test/kotlin/com/ford/internalprojects/peoplemover/role/RoleControllerApiTest.kt]: Yep you are right, fixed this!

File: ui/src/tests/Product.test.tsx [hunk 4]
```diff
@@ -125,73 +125,64 @@ describe('Products', () => {
...
                     {
                         id: 1,
                         productId: 2,
-                        spaceId: 0,
-                        spaceUuid: 'a',
+                        spaceUuid: 'bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb',
                         person: {
                             newPerson: false,
-                            spaceId: 0,
+                            spaceUuid: 'spaceUuid',
```
[the remark this answers is not in the document]
Comment (alexwojtala) [reply ui/src/tests/Product.test.tsx]: Yep, missed that one

**Pull request merged**